rust/operator-binary/src/catalog/hive.rs

```diff
@@ -27,13 +28,15 @@ impl ToCatalogConfig for HiveConnector {
         catalog_namespace: Option<String>,
         client: &Client,
     ) -> Result<CatalogConfig, FromTrinoCatalogError> {
-        let mut config = CatalogConfig::new(catalog_name.to_string(), CONNECTOR_NAME);
+        // Get config that is generated by attributed shared between Hive and Iceberg
+        let mut config = self
+            .common
+            .to_catalog_config(catalog_name, catalog_namespace, client)
+            .await?;
 
-        config.add_configmap_property(
-            "hive.metastore.uri",
-            self.metastore.config_map.clone(),
-            "HIVE",
-        );
+        // As the `HiveAndIcebergCommonAttributes` struct can't know for which connector it is called
+        // we have to overwrite the connector name here
+        config.add_property("connector.name", CONNECTOR_NAME);
```

Comment:
As mentioned before, i would use one config struct for each connector. This can contain the connector name and we wont get a huge "shared" struct where half of the attributes do not matter for different connectors.